nifi-extension-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/ConsumeAzureEventHub.java

```diff
@@ -543,10 +589,35 @@ protected EventProcessorClient createClient(final ProcessContext context) {
 
         if (checkpointStrategy == CheckpointStrategy.AZURE_BLOB_STORAGE) {
             final String containerName = defaultIfBlank(context.getProperty(STORAGE_CONTAINER_NAME).evaluateAttributeExpressions().getValue(), eventHubName);
-            final String storageConnectionString = createStorageConnectionString(context);
-            final BlobContainerClientBuilder blobContainerClientBuilder = new BlobContainerClientBuilder()
-                    .connectionString(storageConnectionString)
-                    .containerName(containerName);
+            final String storageAccountName = context.getProperty(STORAGE_ACCOUNT_NAME).evaluateAttributeExpressions().getValue();
+            final String domainName = getStorageDomainName(serviceBusEndpoint);
+            final BlobStorageAuthenticationStrategy blobStorageAuthenticationStrategy =
+                    context.getProperty(BLOB_STORAGE_AUTHENTICATION_STRATEGY).asAllowableValue(BlobStorageAuthenticationStrategy.class);
+
+            if (blobStorageAuthenticationStrategy == null) {
+                throw new IllegalArgumentException("Blob Storage Authentication Strategy must be specified");
+            }
+
+            final BlobContainerClientBuilder blobContainerClientBuilder = new BlobContainerClientBuilder();
+
+            final Runnable storageCredentialConfigurator = switch (blobStorageAuthenticationStrategy) {
```

Comment:
I'm not following why this is a `Runnable` and then invoked immediately in this same thread, is there a particular reason for this approach?